src/services/tx/txSender.ts

```diff
@@ -57,7 +57,19 @@ export const createTx = async (txParams: SafeTransactionDataPartial, nonce?: num
   // Set the recommended nonce and safeTxGas if not provided
   if (nonce === undefined) {
     const chainId = await safeSDK.getChainId()
-    const estimation = await estimateSafeTxGas(String(chainId), safeSDK.getAddress(), txParams)
+    const safeAddress = await safeSDK.getAddress()
+    let estimation: SafeTransactionEstimation
+    try {
+      estimation = await estimateSafeTxGas(String(chainId), safeSDK.getAddress(), txParams)
+    } catch (e) {
+      try {
+        const minTxParams = { ...txParams, data: '0x', to: safeAddress }
+        estimation = await estimateSafeTxGas(String(chainId), safeSDK.getAddress(), minTxParams)
+      } catch (e) {
+        console.error('Error estimating SafeTx gas', e)
+        estimation = { currentNonce: 0, recommendedNonce: 0, safeTxGas: '0' }
```

Comment:
As discussed, let's not pass `recommendedNonce: 0` here, and instead take the current safe nonce (I think the SDK does it automatically if you pass `nonce: undefined`).

Reply:
It doesn't, hence setting it to either the recommended one or that which is passed as an argument:

![image](https://user-images.githubusercontent.com/20442784/193226618-279d4c26-e876-4f3a-b10c-2a8a5a2cd9c5.png)

In `safe-react` we use 0. I am unsure as to when a rejection estimation could fail though? The issue says we return `0` sometimes though.

Reply:
It was a standard transaction creation. Disregard my comment then.

Reply:
I am not sure how to approach this suggestion. Should I pass the currentSafeNonce to the createTx?
I can't call the hook to retrieve the currentNonce but I need a value in `SafeTransactionEstimation` type